mirrord/layer-win/src/hooks/process/mod.rs

```diff
@@ -1,181 +1,273 @@
+                    environment,
+                    current_directory,
+                    startup_info,
+                    process_information,
+                    restricted_user_token,
+                )
+            };
+
+            if result != 0 {
+                tracing::debug!("Fallback to original API succeeded");
+            } else {
+                tracing::error!("Both unified creation and fallback failed");
+            }
+
+            result
+        }
     }
 }
 
-// https://github.com/winsiderss/systeminformer/blob/f9c238893e0b1c8c82c2e4a3c8d26e871c8f09fe/phnt/include/ntpsapi.h#L1928
-type NtCreateProcessExType = unsafe extern "system" fn(
-    PHANDLE,
-    ACCESS_MASK,
-    PCOBJECT_ATTRIBUTES,
-    HANDLE,
-    ULONG,
-    HANDLE,
-    HANDLE,
-    HANDLE,
-    ULONG,
-) -> NTSTATUS;
-static NT_CREATE_PROCESS_EX_ORIGINAL: OnceLock<&NtCreateProcessExType> = OnceLock::new();
-
-unsafe extern "system" fn nt_create_process_ex_hook(
-    process_handle_ptr: PHANDLE,
-    desired_access: ACCESS_MASK,
-    object_attributes_ptr: PCOBJECT_ATTRIBUTES,
-    parent_process: HANDLE,
-    flags: ULONG,
-    section_handle: HANDLE,
-    debug_port: HANDLE,
-    token_handle: HANDLE,
-    reserved: ULONG,
-) -> NTSTATUS {
-    // Configuration check for future use
-    let _env_enabled = &layer_setup().layer_config().feature.env;
-    let _fs_enabled = &layer_setup().layer_config().feature.fs;
-    let _network_enabled = &layer_setup().layer_config().feature.network;
-
-    // TODO: Implement process creation interception based on configuration
-    // For now, call the original function
-    let original = NT_CREATE_PROCESS_EX_ORIGINAL.get().unwrap();
-    unsafe {
-        original(
-            process_handle_ptr,
-            desired_access,
-            object_attributes_ptr,
-            parent_process,
-            flags,
-            section_handle,
-            debug_port,
-            token_handle,
-            reserved,
-        )
+/// Hook LoadLibraryW for API monitoring during injection
+///
+/// This detour monitors DLL loading to provide comprehensive visibility into
+/// what modules are being loaded during process injection. Useful for debugging
+/// complex injection scenarios and understanding the full API landscape.
+unsafe extern "system" fn loadlibrary_w_detour(lpLibFileName: *const u16) -> HMODULE {
+    // Call the original LoadLibraryW first
+    let original = LOAD_LIBRARY_W_ORIGINAL.get().unwrap();
+    let result = unsafe { original(lpLibFileName) };
+
+    // Log LoadLibrary calls for debugging
+    if !lpLibFileName.is_null() {
+        // Convert the wide string to a Rust string for logging
+        let len = unsafe {
+            (0..).take_while(|&i| *lpLibFileName.offset(i) != 0).count()
+        };
+
+        if len > 0 {
+            let wide_slice = unsafe { std::slice::from_raw_parts(lpLibFileName, len) };
+            let lib_name = String::from_utf16_lossy(wide_slice);
```

Comment:
should be avoided through str-win function, normally here you'd take the slice above and turn it to u16_buffer_to_string